start.py

```diff
@@ -1315,16 +1316,24 @@ def _execute_commands_against_kerberized_service(cluster, commands, service_keyt
     cluster.primary_node.execute(' && '.join(kinit_commands + commands + kdestroy_command))
 
 
-def _configure_after_start(deployment, cluster_name, cluster, quiet, kerberos_principals):
+def _configure_after_start(deployment, cluster_name, cluster, quiet, kerberos_principals, cdh_version_tuple):
     # To contact kerberized HDFS using keytab of any of the Kerberos principals,
     # for each Kerberos principal there needs to be HDFS /user/{principal} directory.
-    if kerberos_principals:
-        cluster_service_types = {service['type']
-                                 for service
-                                 in deployment.get_cluster_services(cluster_name)}
+    cluster_service_types_names = {(service['type'], service['name'])
+                                   for service
+                                   in deployment.get_cluster_services(cluster_name)}
+    cluster_service_types = [service[0] for service in cluster_service_types_names]
```

Comment:
What's with the "service[0]" thing? Kinda hard to read since I didn't know we were just getting some ordered list :)